contrib/startup/GafferUI/layoutCompatibility.py

```diff
@@ -40,23 +40,45 @@
 import GafferUI
 
 import imath
+import inspect
 
-if Gaffer.About.compatibilityVersion() < 54 :
+# The signature for CompoundEditor.__init__ has changed some:
+#  - 0.54.0.0 introduced 'windowState' and 'detachedPanels'
+#  - 0.x.x.x migrated to the opaque _state dict
+# We need to patch the constructor such that it only receives the args that
+# it expects
+
+initSpec = inspect.getargspec( GafferUI.CompoundEditor.__init__ )
+
+def __initWrapper( originalInit ) :
+
+	def init( self, *args, **kwargs ) :
+
+		# If this version uses _state and incoming args are pre-state, nest them
```

Comment:
@andrewkaufman you can rest easy now

Reply:
In an offline discussion we've agreed to split things so that the version that installs with Gaffer itself will deal with backwards compatibility (loading old layouts), and the contrib file will remain for forwards compatibility (because it needs to be installed separately for old gaffer versions).